integration/pages/BasePage.js

```diff
@@ -0,0 +1,52 @@
+const config = require('../utils/config.js');
+const { until } = require('selenium-webdriver');
+
+class BasePage {
+  constructor(driver) {
+    this.driver = driver;
+  }
+
+  async visit(route) {
+    await this.driver.get(config.baseUrl + route);
+  }
+
+  find(locator) {
+    return this.driver.findElement(locator);
+  }
+
+  async findAll(locator) {
+    return await this.driver.findElements(locator);
+  }
+
+  async click(locator) {
+    await this.find(locator).click();
+  }
+
+  async type(locator, inputText) {
+    await this.find(locator).sendKeys(inputText);
+  }
+
+  async isDisplayed(locator, timeout) {
+    if (timeout) {
+      await this.driver.wait(until.elementLocated(locator), timeout);
+      await this.driver.wait(until.elementIsVisible(this.find(locator)), timeout);
+      return true;
+    } else {
+      try {
+        return await this.find(locator).isDisplayed();
+      } catch (error) {
+        return false;
+      }
+    }
+  }
+
+  async isDisappeared(locator, timeout) {
+    return await this.driver.wait(until.stalenessOf(locator), timeout);
```

Comment:
we have to be careful with these, since we'll base all of oru tests on this base methods. Is "stale" the correct method to use here? Are there other indications that a element is no longer on the page?

Reply:
Yes, this is the official condition for this situation.

>Creates a condition that will wait for the given element to become stale. An element is considered stale once it is removed from the DOM, or a new page has loaded.

Reply:
Then we should call this isStale? Dissappeared could mean different things right?

Reply:
Updated.